processor/src/main/kotlin/com/thinkslynk/fabric/annotations/builder/RegistryBuilder.kt

```diff
@@ -0,0 +1,66 @@
+package com.thinkslynk.fabric.annotations.builder
+
+import com.squareup.kotlinpoet.*
+import com.thinkslynk.fabric.annotations.processor.definition.Definition
+import java.nio.file.Path
+
+class RegistryBuilder(packageName: String, className: String, funcName: String, val registry: MemberName) {
+    constructor(packageName: String, className: String, funcName: String, registry: String) : this(
+        packageName,className,funcName, MemberName(registryClass,registry)
+    )
+
+    private val fileBuilder = FileSpec.builder(packageName, className)
+
+    private val objectBuilder = TypeSpec.objectBuilder(className)
+
+    private val registerBuilder = FunSpec.builder(funcName)
+        .addModifiers(KModifier.PUBLIC)
+
+    fun writeTo(folder: Path) {
+        fileBuilder.addType(
+            objectBuilder
+                .addFunction(registerBuilder.build())
+                .build()
+        )
+            .build()
+            .writeTo(folder)
+
+    }
+
+    fun addDefinition(element: Definition) {
+        objectBuilder
+            .addProperty(
+                PropertySpec.builder(element.name, element.type)
+                    .initializer(element.initializer)
+                    .build()
+            )
+        registerBuilder
+            .addStatement(
+                """%T.register(%M, %T(%S, %P), %L)""",
```

Comment:
`%T` (Type) also makes sure I'm actually passing a type there and will automatically add the import.
`%M` (Member) is the same but for Members
`%S` (String) adds `"` around (and escapes `$`)
`%P` (Parametrised String?) is like `%S` but without the escaping
`%N` (Name) Is for referencing a generated thing (by name) (And I should probably be using that instead of `%L` here)
`%L` (Literal) is for literally copying the string passed and is identical to `${stuff}`

I also prefer it as it makes code that can be easily split over multiple lines and doesn't require `\"` although I'm already using `"""` here anyway so I could just use `"`